pkg/rancher-desktop/store/preferences.ts

```diff
@@ -42,31 +42,33 @@ const settingsUri = (port: number) => uri(port, 'settings');
 const lockedUri = (port: number) => uri(port, 'settings/locked');
 
 /**
- * Creates an object composed of active WSL Integrations.
- * @param integrations: The source collection, containing all WSL integrations.
- * @returns Returns a new object, containing only active WSL Integrations.
+ * Normalize WSL integrations configuration.
+ * @param integrations The source collection, containing all WSL integrations.
+ * @param mode How normalization should take place
+ * @returns Returns a new object with normalized WSL configuration.
  */
-const pickWslIntegrations = (integrations: Record<string, boolean>) => {
-  const someVal = Object.fromEntries(
-    Object
-      .entries(integrations)
-      .filter(([_key, val]) => val === true),
-  );
+const normalizeWslIntegrations = (integrations: Record<string, boolean>, mode: 'diff' | 'submit') => {
+  const normalizeFn = {
+    diff:   (entries: [string, boolean][]) => entries.filter(([, v]) => v === true),
+    submit: (entries: [string, boolean][]) => entries.map(([k, v]) => [k, v || null] as const),
+  }[mode];
+  const someVal = Object.fromEntries(normalizeFn(Object.entries(integrations)));
```

Comment:
Can just return the result of `Object.fromEntries(...)` (Same comment for the code that this PR replaces)

Reply:
Yep, totally doable.